app/pages/cif/project-revision/[projectRevision]/form/[formIndex]/index.tsx

```diff
@@ -109,30 +133,34 @@ export function ProjectOverviewForm({
     );
   };
 
-  const taskList = <TaskList projectRevision={query.projectRevision} />;
+  const taskList = (
+    <TaskList projectRevision={query.projectRevision} mode={mode} />
+  );
 
   const handleSubmit = () => {
-    if (existingRevision) {
+    if (mode === "update" || formIndex === formPages.length - 1) {
       router.push(getProjectRevisionPageRoute(query.projectRevision.id));
     } else {
       router.push(
-        getProjectRevisionManagersFormPageRoute(query.projectRevision.id)
+        getProjectRevisionFormPageRoute(query.projectRevision.id, formIndex + 1)
       );
     }
   };
 
+  const EditComponent = formPages[formIndex].editComponent;
+  const ViewComponent = formPages[formIndex].viewComponent;
   return (
     <DefaultLayout session={query.session} leftSideNav={taskList}>
-      {query.projectRevision.changeStatus === "committed" ? (
+      {query.projectRevision.changeStatus === "committed" && ViewComponent ? (
         <>
           {createEditButton()}
-          <ProjectFormSummary
+          <ViewComponent
             projectRevision={query.projectRevision}
             viewOnly={true}
           />
         </>
       ) : (
-        <ProjectForm
+        <EditComponent
           query={query}
           projectRevision={query.projectRevision}
           onSubmit={handleSubmit}
```

Comment:
This means that if for some reason there is no view component (like it's the case for quarterly and annual reports), then we show the edit form even in view mode. It might lead to a bug?

Reply:
@pbastia  Are the report pages not going to get a view only component?

Reply:
They are! Just being mindful of writing code that can handle our potential mistakes or forgetfulness :)

Reply:
I replaced the `viewComponent: null` with `viewComponent: () => null` to avoid people making changes to the reports that aren't being saved anywhere. Good catch, it was rendering the form in edit mode despite being in view mode, so changes could be made but wouldn't get saved anywhere. In the future, if we have components that actually don't have a `viewComponent` we'll have to rethink this maybe.